extensions/wikia/ArticleComments/modules/ArticleCommentsController.class.php

```diff
@@ -24,25 +32,20 @@ public function executeIndex() {
 						if ( !$isMobile ) {
 							$this->wg->Out->redirect( $oTitle->getLocalURL() );
 						} else {
-							$result = array();
-							$canComment = ArticleCommentInit::userCanComment( $result, $oTitle );
-
-							//this check should be done for all the skins and before calling ArticleComment::doPost but that requires a good bit of refactoring
-							//and some design review as the OAsis/Monobook template doesn't handle error feedback from this code
-							if ( $canComment == true ) {
-								if ( empty( $response[2]['error'] ) ) {
+							if ( empty( $response[2]['error'] ) ) {
 									//wgOut redirect doesn't work when running fully under the
 									//Nirvana stack (WikiaMobile skin), also send back to the first page of comments
 									$this->response->redirect( $oTitle->getLocalURL( array( 'page' => 1 ) ) . '#article-comments' );
-								} else {
-									$this->response->setVal( 'error', $response[2]['msg'] );
-								}
 							} else {
-								$this->response->setVal( 'error', $result['msg'] );
+									$this->response->setVal( 'error', $response[2]['msg'] );
 							}
 						}
 					}
 				}
+
+				if ( !$canComment ) {
```

Comment:
It may be cleaner to move this to line 25 and turn the subsequent `if` into an `else if ( $sSubmit && $sComment && $iArticleId )`